src/NServiceBus.Core/CriticalError/CriticalError.cs

```diff
@@ -37,8 +37,8 @@ void DefaultCriticalErrorHandling()
                 return;
             }
 
-            builder.Build<IStartableBus>()
-                .Dispose();
+            builder.Build<IStoppableEndpoint>()
```

Comment:
I still hate this code. Can we come up with a better way to shutdown? This is too much magic and reliable on the correct binding/scope of the bus.